AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSCosmosDb.m

```diff
@@ -132,16 +132,19 @@ + (void)performCosmosDbAsyncOperationWithHttpClient:(MSCosmosDbIngestion *)httpC
                                         tokenResult:(MSTokenResult *)tokenResult
                                          documentId:(NSString *)documentId
                                          httpMethod:(NSString *)httpMethod
-                                               body:(nullable NSData *)body
+                                               body:(NSData *_Nullable)body
+                                  additionalHeaders:(NSDictionary *_Nullable)additionalHeaders
                                   completionHandler:(MSCosmosDbCompletionHandler)completionHandler {
-
-  [MSCosmosDb performCosmosDbAsyncOperationWithHttpClient:httpClient
-                                              tokenResult:tokenResult
-                                               documentId:documentId
-                                               httpMethod:httpMethod
-                                                     body:body
-                                        additionalHeaders:nil
-                                        completionHandler:completionHandler];
+  [MSCosmosDb
```

Comment:
why do we need 2 performCosmosDbAsyncOperationWithHttpClient:httpClient with different completion handlers?

can it just be one completion handler which has all 3 arguments?
also, the naming `completionHandlerWithHeaders` is awkward.

Reply:
We can have reduce it to one, since it's internal.  It was easier to avoid conflicts when we wrote it.

I actually like the longer name because it makes the file a lot more readable because of how our formatting works.  Code below the callback is indented 2 spaces after the first character of the last argument so having a longer name moves it over significantly.  I don't like the convention but it is what it is

Reply:
It was done this way also to avoid having to add `__unused` all over the place. We can probably merge now.